### ISSUE TITLE
Add more functions to tinymce in admin panel

### ISSUE THREAD

**Issue by SharakPL** [PrestaShop/PrestaShop#12417]
This may seem like a good enough set of functions:

![obraz](https://user-images.githubusercontent.com/5007456/52180518-8b545800-27e7-11e9-874b-6ffd67195473.png)

but "**Clear formatting**" function is also **a must**. Most of the time content is copy-pasted from some docs and it brings all the styling set in these docs. In prestashop 1.7 there's no way to clear this so you have to paste plain text and do html structure all over again.
Please add **Clear formatting** option 🙏 

Other tinymce functions I often look for while editing content is:
- text color
- font size

**Comment by khouloudbelguith** [comment]
Hi @SharakPL,

Thanks for your suggestion.
It could an improvement issue for the PS1.7.
Would you be able to do a PR for this improvement?
@colinegin @marionf what do you think of his improvement?
Thanks!

**Comment by colinegin** [comment]
@eternoendless and @PrestaShop/prestashop-core-developers what do you think of this suggestion ? Is it technically complex ?

**Comment by jolelievre** [comment]
Hi @SharakPL
did you try to paste your text using `Ctrl + Shift + V` (or `Cmd + Shift + V` on Mac)
It copies your texte as raw texte without any formatting (also works in Gmail), it prevents you
from copying your text in texteditor (Notepad and such) It actually saves me a lot of time

Maybe this could solve your issue? (or at least provide a temporary solution)

**Comment by SharakPL** [comment]
@jolelievre you're saviour. Thanks!
Still at least a popup with this shortcut and some other useful would be helpful...

**Comment by jolelievre** [comment]
@SharakPL you're welcome! ;)
Of course it's just a tip, it doesn't close this discussion nor this issue.

To answer your question @colinegin it shouldn't be complex to modify it, it is only a config in tinymce: https://www.tiny.cloud/docs/configure/editor-appearance/#toolbarn
The question would be more about a reflexion of which buttons/features we want to add, here is the list of available controls: https://www.tiny.cloud/docs/advanced/editor-control-identifiers/#toolbarcontrols

Or a more advanced modif would be to allow users to customize their editor configuration.

**Comment by PierreRambaud** [comment]
Before adding features to TinyMce, we need to use the latest version, with npm..  :roll_eyes:

**Comment by SharakPL** [comment]
@jolelievre unfortunately `Ctrl + Shift + V` doesn't work good enough as it strips also all HTML tags and leaves only 1 paragraph and some breaks. Turns out TinyMce's Clear formatting function is needed to clean only the styles and leave all separate paragraphs, headings, lists etc.

**Comment by ghost** [comment]
Hi @SharakPL 
Thanks for the feedback ! That's a good idea.
We will look into which Release roadmap we will be able to add it.
We won't be able to do it for the 1.7.6, as Pierre said we have to update our dependencies first and it will be done later.

We might use this time to ask the community more globally which features they would like to see in the TinyMce editor. Stay tuned 🙂

For the moment, I can recommend you to paste your text in a basic text editor such as Notepad or Sublime Text.
It removes all styling while keeping html tags.

**Comment by SharakPL** [comment]
> For the moment, I can recommend you to paste your text in a basic text editor such as Notepad or Sublime Text.

It's easier to paste directly into tinymce editor with `CTRL+SHIFT+V` like @jolelievre suggested as you get the same effect and skip the notepad step, but it's still PITA to do it this way if you have a lot of content to format

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team what's the status of this issue?

**Comment by SharakPL** [comment]
@hibatallahAouadni what is @PrestaShop/product-team? I get 404

**Comment by khouloudbelguith** [comment]
Hi @SharakPL,

It is only displayed to our member team.
This link is provided to ping our product team :slightly_smiling_face: 

Thank you!

**Comment by SharakPL** [comment]
@khouloudbelguith thank you.
@github It should be some kind of _access restricted_ message instead, because 404 is rather misleading. Looks like wrong link target.

**Comment by SharakPL** [comment]
Yes, it's still relevant. More formatting options are required. Also typical Bootstrap buttons like `btn-primary` `btn-outline-primary`  `btn-secondary` `btn-info` etc. The editor offers only a `button` as a style for the link.

**Comment by boo-code** [comment]
The version objection does not apply to `removeformat`. The vendored TinyMCE is 4.9.11 and its modern theme already registers the button itself - `removeformat:["Clear formatting","RemoveFormat"]` fed into `addButton(t,{tooltip:e[0],cmd:e[1]})` in `js/tiny_mce/themes/modern/theme.min.js` - and the mobile theme has its own entry in its item registry. So the control exists in the version core ships; it was never listed in a toolbar. No plugin, no dependency, no upgrade.

`git grep toolbar1` returns two files: `admin-dev/themes/new-theme/js/components/tinymce-editor.js` for migrated pages and `js/admin/tinymce.inc.js` for the legacy back-office theme. Each has a desktop `toolbar1` and a separate `mobile.toolbar`, and both merge `window.defaultTinyMceConfig` afterwards, which nothing in core sets - so those four strings are every default toolbar in the back office.

Fix prepared against `9.2.x`: the button added to all four, between `strikethrough` and `blockquote`. Measured by loading the vendored `tinymce.min.js` with the `prestashop` skin and the exact toolbar strings from before and after, then reading the rendered `aria-label` of every `.mce-toolbar .mce-btn` - 16 buttons before, 17 after, the addition being "Clear formatting". Functional check on the command with content carrying `font-family`, `color`, `font-size` and `background` styles plus `<b>` and `<i>`:

```
<h2>Heading</h2>
<p>Bold word and italic</p>
<ul>
<li>one</li>
<li>two</li>
</ul>
```

Zero `style=` attributes and no `<b>`/`<i>`/`<span>`/`<font>` left, with the `h2`, the `p` and both `li` intact. That is the difference from `Ctrl+Shift+V`, which flattens the structure as well - the point made in the 2019-02-11 comment.

Text colour is already there and is not part of the change: core ships its own `colorpicker` plugin at `js/tiny_mce/plugins/colorpicker/plugin.min.js` which registers a "Color Picker" toolbar button, and that button renders in the measurement above.

What is left open, and what the change deliberately does not touch: font size (`fontsizeselect` is registered by the same vendored theme, so it is also a one-token addition), a style dropdown, and the Bootstrap button classes asked for in March. Those are the "which buttons do we want" list, and they need their own issue rather than being closed along with this one.
